modules/snowflake-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/snowflake/Snowflake.scala

```diff
@@ -100,29 +118,63 @@ object Snowflake {
+                    val stageName = tgt.folderMonitoringStage.getOrElse(throw new IllegalStateException("Folder Monitoring is launched without monitoring stage being provided"))
+                    Fragment.const0(s"@$schema.$stageName/${source.folderName}")
+                  case _: LoadAuthMethod.TempCreds =>
+                    Fragment.const0(source)
+                }
+                val frCredentials = loadAuthMethodFragment(loadAuthMethod)
+                sql"""|COPY INTO $frTableName
+                      |FROM $frPath $frCredentials
+                      |FILE_FORMAT = (TYPE = CSV)""".stripMargin
+
+              case Statement.EventsCopy(path, _, columns, _, typesInfo, _) =>
                 // This is validated on config decoding stage
-                val stageName = monitoringStage.getOrElse(throw new IllegalStateException("Folder Monitoring is launched without monitoring stage being provided"))
-                val frPath      = Fragment.const0(s"@$schema.$stageName/${source.folderName}")
-                sql"COPY INTO $frTableName FROM $frPath FILE_FORMAT = (TYPE = CSV)"
-
-              case Statement.EventsCopy(path, _, columnsToCopy, _, typesInfo, _) => {
-                def columnsForCopy: String = columnsToCopy.names.map(_.value).mkString(",") + ",load_tstamp"
-                def columnsForSelect: String = columnsToCopy.names.map(c => s"$$1:${c.value}").mkString(",") + ",current_timestamp()"
-
-                val frPath = Fragment.const0(s"@${qualify(stage)}/${path.folderName}/output=good/")
-                val frCopy = Fragment.const0(s"${qualify(EventsTable.MainName)}($columnsForCopy)")
-                val frSelectColumns = Fragment.const0(columnsForSelect)
-                val frOnError = Fragment.const0(s"ON_ERROR = ${onError.asJson.noSpaces}")
+                val stageName = tgt.transformedStage.getOrElse(throw new IllegalStateException("Transformed stage is tried to be used without being provided"))
+                val frPath = Fragment.const0(s"@${qualify(stageName)}/${path.folderName}/output=good/")
+                val frCopy = Fragment.const0(s"${qualify(EventsTable.MainName)}(${columnsForCopy(columns)})")
+                val frSelectColumns = Fragment.const0(columnsForSelect(columns))
+                val frOnError = Fragment.const0(s"ON_ERROR = ${tgt.onError.asJson.noSpaces}")
                 val frFileFormat = buildFileFormatFragment(typesInfo)
                 sql"""|COPY INTO $frCopy
                       |FROM (
                       |  SELECT $frSelectColumns FROM $frPath
                       |)
                       |$frFileFormat
                       |$frOnError""".stripMargin
-              }
+
+              case Statement.CreateTempEventTable(table) =>
+                val frTableName = Fragment.const(qualify(table))
+                val frTableColumn = Fragment.const(TempTableColumn)
+                sql"CREATE TEMPORARY TABLE IF NOT EXISTS $frTableName ( $frTableColumn OBJECT )"
```

Comment:
I can't think of a case it might exist already. I just put it there just in case but I agree now it would be better to remove it if we don't expect any case like that. Also, it would be better to get error instead of silently continue in case like that. I will remove it 👍